Mistica/Source/Components/Callout/Internals/CalloutContentBase.swift

```diff
@@ -0,0 +1,178 @@
+    static let iconLayoutMargins = NSDirectionalEdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 0)
+    static let closeImageTopMargin = CGFloat(0)
+    static let closeImageTralingMargin = CGFloat(-12)
+    static let closeImageWidth = CGFloat(24)
+    static let closeImageHeight = CGFloat(24)
+    static let iconWidth = CGFloat(24)
+}
+
+/// CalloutContentBase is a horizontal stackview where we will have the image - message&buttons - close button
+class CalloutContentBase: UIStackView {
+    let calloutBaseView = CalloutTitleActions()
+
+    private let iconContainerView = UIView()
+    private var iconImageView = CalloutAsset()
+    private let closeImageView = IntrinsictImageView()
+    private let onCloseButtonTap: () -> Void
+
+    init(onCloseButtonTap: @escaping () -> Void) {
+        self.onCloseButtonTap = onCloseButtonTap
+
+        super.init(frame: .zero)
+
+        commomInit()
+    }
+
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: Internal
+
+extension CalloutContentBase {
+    var title: String? {
+        get {
+            calloutBaseView.contentView.title
+        }
+        set {
+            calloutBaseView.contentView.title = newValue
+        }
+    }
+
+    var descriptionTitle: String? {
+        get {
+            calloutBaseView.contentView.descriptionTitle
+        }
+        set {
+            calloutBaseView.contentView.descriptionTitle = newValue
+        }
+    }
+
+    override var accessibilityElements: [Any]? {
+        get {
+            [
+                calloutBaseView.buttonsView
+            ].compactMap { $0 }
+        }
+        set {}
+    }
+
+    var iconContentMode: UIView.ContentMode {
+        get {
+            iconImageView.contentMode
+        }
+        set {
+            iconImageView.contentMode = newValue
+        }
+    }
+
+    func configure(withConfiguration configuration: CalloutConfiguration) {
+        if configuration.asset != .none {
+            if iconContainerView.superview == nil {
+                insertArrangedSubview(iconContainerView, at: 0)
+                setCustomSpacing(Constants.spacingAfterIconView, after: iconContainerView)
+            }
+            iconImageView.assetType = configuration.asset
+        } else {
+            iconContainerView.removeFromSuperview()
+            iconImageView.assetType = .none
+        }
+
+        if !configuration.canClose {
+            closeImageView.removeFromSuperview()
+        } else {}
+
+        calloutBaseView.title = configuration.title
+        calloutBaseView.descriptionTitle = configuration.descriptionTitle
+
+        switch configuration.actions {
+        case .primary(let primaryButton):
+            calloutBaseView.configureButtons(primaryButton: primaryButton)
+        case .primaryAndLink(primary: let primaryButton, link: let linkButton):
```

Comment:
This is just personal preference, but you can avoid writing let twice and you can remove the labels so that it reads a bit more succinct:

```suggestion
        case let .primaryAndLink(primaryButton, linkButton):
```